src/store/metadata/oxigraph/store.py

```diff
@@ -102,15 +112,95 @@ def get_editions(self, dataset_id: str) -> Optional[Dict]:  # pragma: no cover
         """
         Gets all editions of a specific dataset
         """
-        raise NotImplementedError
+        # Populate the graph from the database
+        graph = self.db
+
+        # Use the construct wrappers to pull the raw RDF triples
+        # (as one rdflib.Graph() for each function) and add them
+        # together to create a single Graph of the
+        # data we need.
+        result: Graph = construct_editions(graph, dataset_id)
+
+        # Serialize the graph into jsonld
+        data = json.loads(result.serialize(format="json-ld"))
+
+        # Use a context file to shape our jsonld, removing long form references
+        data = jsonld.flatten(
+            data, {"@context": constants.CONTEXT, "@type": "hydra:Collection"}
+        )
+        editions_graph = _get_single_graph_for_field(data, "@type")
```

Comment:
check if its None and return None if it is please, gotta guard against key errors later on.